src/main/java/formflow/library/data/SubmissionRepositoryService.java

```diff
@@ -33,14 +32,14 @@ public SubmissionRepositoryService(SubmissionRepository repository, SubmissionEn
    * @param submission the submission to save, not null
    * @return UUID of the saved submission
    */
-  public UUID save(Submission submission) {
+  public Submission save(Submission submission) {
     var newRecord = submission.getId() == null;
-    UUID id = repository.save(encryptionService.encrypt(submission)).getId();
+    Submission savedSubmission = repository.save(encryptionService.encrypt(submission));
     if (newRecord) {
-      log.info("created submission id: " + id);
+      log.info("created submission id: " + savedSubmission.getId());
     }
-    submission.setId(id);
-    return id;
+    // straight from the db will be encrypted, so decrypt first.
+    return encryptionService.decrypt(savedSubmission);
   }
```

Comment:
Note that `save` now returns the Submission versus the id (as before).  This helped with some test code challenges and seems like a good thing to do overall.   Underneath, in the JPA, the object is either persisted or merged. If merged, the underlying save will return a different object. It seemed okay and useful to have this save return an object as well.